plugin/builtin/patchtransformer/PatchTransformer.go

```diff
@@ -64,51 +65,69 @@ func (p *plugin) Config(
 			"unable to parse SM or JSON patch from [%v]", p.Patch)
 	}
 	if errSM == nil {
-		p.loadedPatch = patchSM
-		if p.Options["allowNameChange"] {
-			p.loadedPatch.AllowNameChange()
-		}
-		if p.Options["allowKindChange"] {
-			p.loadedPatch.AllowKindChange()
+		p.smPatches = patchesSM
+		for _, loadedPatch := range p.smPatches {
+			if p.Options["allowNameChange"] {
+				loadedPatch.AllowNameChange()
+			}
+			if p.Options["allowKindChange"] {
+				loadedPatch.AllowKindChange()
+			}
 		}
 	} else {
-		p.decodedPatch = patchJson
+		p.jsonPatch = patchesJson
 	}
 	return nil
 }
 
 func (p *plugin) Transform(m resmap.ResMap) error {
-	if p.loadedPatch == nil {
-		return p.transformJson6902(m, p.decodedPatch)
+	if p.smPatches == nil {
+		return p.transformJson6902(m)
 	}
 	// The patch was a strategic merge patch
-	return p.transformStrategicMerge(m, p.loadedPatch)
+	return p.transformStrategicMerge(m)
 }
 
-// transformStrategicMerge applies the provided strategic merge patch
-// to all the resources in the ResMap that match either the Target or
-// the identifier of the patch.
-func (p *plugin) transformStrategicMerge(m resmap.ResMap, patch *resource.Resource) error {
-	if p.Target == nil {
-		target, err := m.GetById(patch.OrgId())
+// transformStrategicMerge applies each loaded strategic merge patch
+// to the resource in the ResMap that matches the identifier of the patch.
+// If only one patch is specified, the Target can be used instead.
+func (p *plugin) transformStrategicMerge(m resmap.ResMap) error {
+	for _, patch := range p.smPatches {
+		if p.Target == nil {
+			target, err := m.GetById(patch.OrgId())
+			if err != nil {
+				return fmt.Errorf("unable to find patch target %s: %w", patch.OrgId().Name, err)
+			}
+			if err := target.ApplySmPatch(patch); err != nil {
+				return fmt.Errorf("%w", err)
+			}
+			continue
+		} else if len(p.smPatches) > 1 {
+			// detail: https://github.com/kubernetes-sigs/kustomize/issues/5049#issuecomment-1440604403
+			return fmt.Errorf("Multiple Strategic-Merge Patch in 'patches' is not allowed to set 'patches.target' field.")
```

Comment:
I think this error message is a bit unclear, partly because there is not really a `patches.target` field, only a `patches[idx].target` field.

In the following case, the first entry is fine, only the second entry is not allowed, but to me the error message sounds like this config would be invalid even if the second entry had `path: other-file-with-one-document.yaml`

```yaml
patches:
- path: file-with-one-document.yaml
  target:
    name: name1
- path: file-with-multiple-documents.yaml
  target:
    name: name2
```

Reply:
I update message at https://github.com/kubernetes-sigs/kustomize/pull/5194/commits/f53483164b2151ed7792c3ff0145428eb6012ad7